ament_cmake_ros/ament_cmake_ros-extras.cmake.in

```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# copied from ament_cmake_ros/ament_cmake_ros-extras.cmake
+# generated from ament_cmake_ros/ament_cmake_ros-extras.cmake.in
 
 include("${ament_cmake_ros_DIR}/build_shared_libs.cmake")
+add_definitions(-DROS_PACKAGE_NAME=\"${PROJECT_NAME}\")
```

Comment:
Out of curiosity: why do the quotes need to be escaped when the overall argument is not wrapped in quotes?

Reply:
Should this be `PROJECT_NAME` or the name extracted from the `package.xml`? Does ament assert that they are the same? In either case might the package xml name make more sense?

Reply:
I don't see the reason myself; I was copying https://github.com/ros/ros_comm/commit/a6a6f7193d42c4d128f381c2ed6d4d3c9b9f384c, but maybe it was just to not assume anything about what is or isn't in the project name..?